convert2rhel/checks.py

```diff
@@ -706,106 +705,122 @@ def is_loaded_kernel_latest():
-        if unsupported_skip == "1":
-            logger.warning(
-                "Detected 'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' environment variable, we will skip "
-                "the %s comparison.\n"
-                "Beware, this could leave your system in a broken state. " % package_to_check
-            )
-        else:
+    # If repoquery output is empty, then something went wrong, we need to
+    # decide wether to bail out or output a warning (only if the user used the
+    # special environment variable for it.
+    if len(repoquery_output) <= 0:
+        unsupported_skip = os.environ.get("CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK", None)
+        if not unsupported_skip:
             logger.critical(
                 "Could not find any %s from repositories to compare against the loaded kernel.\n"
                 "Please, check if you have any vendor repositories enabled to proceed with the conversion.\n"
                 "If you wish to ignore this message, set the environment variable "
                 "'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' to 1." % package_to_check
             )
 
+        logger.warning(
+            "Detected 'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' environment variable, we will skip "
+            "the %s comparison.\n"
+            "Beware, this could leave your system in a broken state. " % package_to_check
+        )
+        return
+
+    packages = []
+    # We are expecting an repoquery output to be similar to this:
+    #   C2R     1671212820      3.10.0-1160.81.1.el7    updates
+    # We need the `C2R` identifier to be present on the line so we can know for
+    # sure that the line we are working with is the a line that contains
+    # relevant repoquery information to our check, otherwise, we just log the
+    # information as debug and do nothing with it.
+    for line in repoquery_output.split("\n"):
+        if line.strip() and "C2R" in line:
+            _, build_time, latest_kernel, repoid = tuple(str(line).split("\t"))
+            packages.append((build_time, latest_kernel, repoid))
+        else:
+            # Mainly for debugging purposes to see what is happening if we got
+            # anything else that does not have the C2R identifier at the start
+            # of the line.
+            logger.debug("Got a line without the C2R identifier: %s" % line)
+
+    # Empty packages list, that means that repoquery probably failed to find
+    # the kernel (or kernel-core) packages, let's just print a warning and
+    # return.
+    if not packages:
+        logger.warning(
+            "Could not find any %s in the available repositories. "
+            "This could mean that the repositories are not enabled, acessible, "
+            "or missing some important information to progress.\n"
+            "Skipping this check as we can't progress with further verification." % package_to_check
+        )
+        return
```

Comment:
Removing this to reuse the if condition above to use the length of the packages instead of the length of repoquery output.